uniffi_bindgen/src/interface/object.rs

```diff
@@ -494,6 +492,7 @@ impl From<uniffi_meta::ConstructorMetadata> for Constructor {
             throws: meta.throws.map(Into::into),
             checksum_fn_name,
             checksum: meta.checksum,
+            self_type,
```

Comment:
I like this change and I think we could take it even further.  If all type-like structs contain a field like `type_`, `self_type` , `ty`, etc.  Then `AsType` can be reduced to `AsRef<Type>`.  Similarly, we could define a `Callable` struct and then change the `Callable` trait to `AsRef<Callable>`.

I'm just mentioning this as food for thought, I don't think we need to do that in this PR.